### PR TITLE
Dr 1830 query test

### PR DESCRIPTION
**Pull request by tcjiang9**

### CODE REVIEW

File: src/main/java/bio/terra/service/search/SearchService.java
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/search/SearchService.java]: Was this changed to support mocking? You could also mock `SearchHits.iterator()` rather than change this code, as it's not obvious from looking at this code why `getHits()` is used instead of the iterator interface.

File: src/main/java/bio/terra/service/search/SearchService.java
Comment (pshapiro4broad) [inline src/main/java/bio/terra/service/search/SearchService.java:158]: I'm wondering why you changed this from `var`. I think we should use `var` but I'd like to find out more about when people think it would be good or bad to use it.

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: I'm assuming that the index related code is still WIP on Florian's branch and is not part of this review?

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: In test code it's OK to static import test classes, and it's usual to static import the mock and matcher classes. So this could be 
```suggestion
        GetAliasesResponse mockResponse = mock(GetAliasesResponse.class);
```

And below you could use `any()` instead of `Mockito.any()`, etc.

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: Does the set have to be modifiable? If not you can use the empty set:
```suggestion
        when(mockResponse.getAliases()).thenReturn(Map.of("0f14d0ab-9605-4a62-a9e4-5ed26688389b", Set.of()));
```

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: Just like `Map.of()`, there's `List.of()` that is IMO a more idiomatic form than `Arrays.asList()`
```suggestion
        List<UUID> snaptshotIdsToQuery = List.of(
```

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: I think you can use the no-arg `any()` here, which can infer the class type based on context.

**Review by pshapiro4broad — APPROVED** [review]
Only a few minor comments, looks good otherwise.

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (fboulnois) [reply src/test/java/bio/terra/service/search/SearchServiceTest.java]: Yes, this is before the rebase I did earlier today.

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (tcjiang9) [reply src/test/java/bio/terra/service/search/SearchServiceTest.java]: In my experience it's better to be explicit about the code where you can, I've been discouraged from using blank any() when possible in the past

File: src/main/java/bio/terra/service/search/SearchService.java
Comment (tcjiang9) [reply src/main/java/bio/terra/service/search/SearchService.java:158]: I think we should use var if we're immediately passing the variable outside the class, or the variable name is descriptive of the type, but if there's still logic we need to perform on it, we should keep the explicit type for readability purposes.

Comment (tcjiang9) [reply src/main/java/bio/terra/service/search/SearchService.java:158]: http://openjdk.java.net/projects/amber/LVTIstyle.html

describes some good var uses. Like if your variable name is just a repeat of the object type. Like `var inputStream` instead of `InputStream inputStream`

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (fboulnois) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: You'll probably want to pull `Mockito.eq` too

File: src/main/java/bio/terra/service/search/SearchService.java
Comment (pshapiro4broad) [reply src/main/java/bio/terra/service/search/SearchService.java:158]: Interesting, I hadn't seen that before, I'll have to read that. The document I'd seen before is https://openjdk.java.net/projects/amber/LVTIFAQ.html

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (fboulnois) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: nit:
```suggestion
        List<UUID> snapshotIdsToQuery = List.of(
```

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (fboulnois) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: Is this the name of the index alias? If so, it should be `idx-<id>` (see `indexName`), correct?

Comment (pshapiro4broad) [inline src/test/java/bio/terra/service/search/SearchServiceTest.java]: It would be good to use a named constant or local variable for `"0f14d0ab-9605-4a62-a9e4-5ed26688389b"` so it's clear that the same UUID is used both here and on line 98.

**Review by fboulnois — CHANGES_REQUESTED** [review]
I don't have a strong feeling on the use of `var`. A few other suggestions:

**Review by pshapiro4broad — APPROVED** [review]
One last (minor) comment

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (tcjiang9) [reply src/test/java/bio/terra/service/search/SearchServiceTest.java]: it doesn't actually affect the test, but I can change it to be more accurate to indexing behavior.

File: src/test/java/bio/terra/service/search/SearchServiceTest.java
Comment (pshapiro4broad) [reply src/test/java/bio/terra/service/search/SearchServiceTest.java]: In the case of matching a method call argument, I don't see the benefit of passing in the parameter type. If you're reading the test code, does having it provide information that makes it easier to understand what the code is doing?

When the method is overloaded the return type of `any()` is ambiguous and must be provided. But I think it's ok to omit it in other cases.

**Review by fboulnois — APPROVED** [review]
lgtm, merge after I merge mine?

**Pull request merged**
